tests/api.c

```diff
@@ -3958,6 +3967,866 @@ static int test_wc_RipeMdFinal (void)
+                if (XMEMCMP(sha3.t, msgCmp, msgCmplen) != 0) {
+                    ret = SSL_FATAL_ERROR;
+                }
+            }
+            /* Pass bad args. */
+            if (ret == 0) {
+                ret = wc_Sha3_384_Update(NULL, msg2, msg2len);
+                if (ret == BAD_FUNC_ARG) {
+                    ret = wc_Sha3_384_Update(&sha3, NULL, 5);
+                }
+                if (ret == BAD_FUNC_ARG) {
+                    wc_Sha3_384_Free(&sha3);
+                    if (wc_InitSha3_384(&sha3, HEAP_HINT, devId)) {
+                        return ret;
+                    }
+                    ret = wc_Sha3_384_Update(&sha3, NULL, 0);
+                    if (ret == 0) {
+                        ret = wc_Sha3_384_Update(&sha3, msg2, msg2len);
+                    }
+                    if (ret == 0 && XMEMCMP(msg2, sha3.t, msg2len) != 0) {
+                        ret = SSL_FATAL_ERROR;
+                    }
+                }
+            }
+            wc_Sha3_384_Free(&sha3);
+
+            printf(resultFmt, ret == 0 ? passed : failed);
+        }
+    #endif /* SHA3_384 */
+
+    #if !defined(WOLFSSL_NOSHA3_512)
+        if (ret == 0) {
+            printf(testingFmt, "wc_Sha3_512_Update()");
+
+            ret = wc_InitSha3_512(&sha3, HEAP_HINT, devId);
+            if (ret != 0) {
+                return ret;
+            }
+            ret = wc_Sha3_512_Update(&sha3, msg, msglen);
+            if (XMEMCMP(msg, sha3.t, msglen) || sha3.i != msglen) {
+                ret = SSL_FATAL_ERROR;
+            }
+            if (ret == 0) {
+                ret = wc_Sha3_512_Update(&sha3, msg2, msg2len);
+                if (XMEMCMP(sha3.t, msgCmp, msgCmplen) != 0) {
+                    ret = SSL_FATAL_ERROR;
+                }
+            }
+            /* Pass bad args. */
+            if (ret == 0) {
+                ret = wc_Sha3_512_Update(NULL, msg2, msg2len);
+                if (ret == BAD_FUNC_ARG) {
+                    ret = wc_Sha3_512_Update(&sha3, NULL, 5);
+                }
+                if (ret == BAD_FUNC_ARG) {
+                    wc_Sha3_512_Free(&sha3);
+                    if (wc_InitSha3_512(&sha3, HEAP_HINT, devId)) {
+                        return ret;
+                    }
+                    ret = wc_Sha3_512_Update(&sha3, NULL, 0);
+                    if (ret == 0) {
+                        ret = wc_Sha3_512_Update(&sha3, msg2, msg2len);
+                    }
+                    if (ret == 0 && XMEMCMP(msg2, sha3.t, msg2len) != 0) {
+                        ret = SSL_FATAL_ERROR;
+                    }
+                }
+            }
+            wc_Sha3_512_Free(&sha3);
+            printf(resultFmt, ret == 0 ? passed : failed);
+        }
+    #endif /* SHA3_512 */
+#endif /* WOLFSSL_SHA3 */
+    return ret;
+
+} /* END testing_wc_Sha3_Update */
+
+/*
+ *  Testing wc_Sha3_224_Final()
+ */
+static int test_wc_Sha3_224 (void)
```

Comment:
Should this function be named "test_wc_Sha3_224_Final()" to match your naming scheme?